### PR TITLE
[BugFix] Fix wrong column order

### LINKED ISSUE #6028: query error from HUDI external table
<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
1. create hudi external table:
```
create external table ex_hudi_tbl8 (
    uuid int
    ,col_boolean boolean
    ,col_int int
    ,col_long bigint
    ,col_float float
    ,col_double double
    ,col_decimal decimal(38,18)
    ,col_date date
    ,col_string string
    ,col_binary varchar(200))
ENGINE=hudi
properties (
    "resource" = "hudi_emr_tn",
    "table" = "hudi_par_parquet_snappy",
    "database" = "hudi_db");
```

2. select from hudi external table error:
```
mysql> select * from ex_hudi_tbl8;
ERROR 1064 (HY000): build chunk meta error
mysql> select * from ex_hudi_tbl8;
+------+-------------+---------+----------+-----------+------------+-----------------------------------------+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------+---------------------+----------------------------------+--------------------+------------------------+------------------------------------------------------------------------------+
| uuid | col_boolean | col_int | col_long | col_float | col_double | col_decimal                             | col_date   | col_string                                                                                                                                                                                                                                                                                                                | col_binary             | _hoodie_commit_time | _hoodie_partition_path           | _hoodie_record_key | _hoodie_commit_seqno   | _hoodie_file_name                                                            |
+------+-------------+---------+----------+-----------+------------+-----------------------------------------+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------+---------------------+----------------------------------+--------------------+------------------------+------------------------------------------------------------------------------+
|    5 |           1 |       5 |        5 |     5.005 |     5.0005 |          50000000000.000000000500000000 | 2020-05-01 | A total of 4,685 companies have been listed on the A-share market in China as of Dec 31, 2021, with 46 percent of them based in Beijing, Shanghai, Shenzhen, Hangzhou, Suzhou, Guangzhou, Ningbo, Nanjing, Wuxi, and Chengdu, said a report of National Business Daily on Wednesday.                                      | 1110001010101011001001 | 20220328173350281   | col_date=2020-05-01/col_int=5    | uuid:5             | 20220328173350281_0_9  | 96802430-c948-4ae1-8291-1d119ecb237b-0_0-684-648_20220328173350281.parquet   |
|   10 |           0 |      10 |       10 |     10.01 |    10.0001 |         100000000000.000000000100000000 | 2020-10-01 | Volkswagen deliveries hit by chip shortages                                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173451281   | col_date=2020-10-01/col_int=10   | uuid:10            | 20220328173451281_0_21 | 8b000532-aeac-44ec-9eb1-0f8893c3b14f-0_0-979-1206_20220328173451281.parquet  |
|    3 |           1 |       4 |        3 |     3.003 |     3.0003 |          30000000000.000000000300000000 | 2020-03-01 | Nation raises caution on overseas packages                                                                                                                                                                                                                                                                                | 1110001010101011001001 | 20220328173335305   | col_date=2020-03-01/col_int=4    | uuid:3             | 20220328173335305_0_12 | e1396e6d-c381-4840-b63d-b382d537df3c-0_0-597-544_20220328173335305.parquet   |
|    8 |           0 |       9 |        8 |     8.008 |     8.0008 |          80000000000.000000000800000000 | 2020-08-01 | Citi report finds MNC mood in China buoyant                                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173433445   | col_date=2020-08-01/col_int=9    | uuid:8             | 20220328173433445_0_19 | 7ed092b8-cb56-4c8c-8ab3-b98688e42ea1-0_0-892-1009_20220328173433445.parquet  |
|    4 |           0 |       5 |        4 |     4.004 |     4.0004 |          40000000000.000000000400000000 | 2020-04-01 | Shanghai, Shenzhen register most newly listed firms in 2021                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173344917   | col_date=2020-04-01/col_int=5    | uuid:4             | 20220328173344917_0_14 | db63d32b-4d3b-4423-abeb-560ba0ee7a09-0_0-655-612_20220328173344917.parquet   |
|    9 |           1 |       9 |        9 |     9.009 |     9.0009 |          90000000000.000000000900000000 | 2020-09-01 | Xi: China, Russia major champions of multilateralism, global justice                                                                                                                                                                                                                                                      | 1110001010101011001001 | 20220328173438751   | col_date=2020-09-01/col_int=9    | uuid:9             | 20220328173438751_0_20 | 974b6e3d-97e7-45aa-a851-76c41411a75a-0_0-921-1072_20220328173438751.parquet  |
|    4 |           0 |       4 |        4 |     4.004 |     4.0004 |          40000000000.000000000400000000 | 2020-04-01 | Shanghai, Shenzhen register most newly listed firms in 2021                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173339621   | col_date=2020-04-01/col_int=4    | uuid:4             | 20220328173339621_0_13 | 85989879-2aad-41c5-8b32-ae6fbe701bd9-0_0-626-577_20220328173339621.parquet   |
|    9 |           1 |      10 |        9 |     9.009 |     9.0009 |          90000000000.000000000900000000 | 2020-09-01 | Xi: China, Russia major champions of multilateralism, global justice                                                                                                                                                                                                                                                      | 1110001010101011001001 | 20220328173444514   | col_date=2020-09-01/col_int=10   | uuid:9             | 20220328173444514_0_12 | 4c4db434-ba10-4344-9612-ad12288b2aa7-0_0-950-1138_20220328173444514.parquet  |
|    5 |           1 |       6 |        5 |     5.005 |     5.0005 |          50000000000.000000000500000000 | 2020-05-01 | A total of 4,685 companies have been listed on the A-share market in China as of Dec 31, 2021, with 46 percent of them based in Beijing, Shanghai, Shenzhen, Hangzhou, Suzhou, Guangzhou, Ningbo, Nanjing, Wuxi, and Chengdu, said a report of National Business Daily on Wednesday.                                      | 1110001010101011001001 | 20220328173355888   | col_date=2020-05-01/col_int=6    | uuid:5             | 20220328173355888_0_15 | 1835a3f2-ec29-4dfd-9452-0e490cd71032-0_0-713-686_20220328173355888.parquet   |
|   10 |           0 |      11 |       10 |     10.01 |    10.0001 |         100000000000.000000000100000000 | 2020-10-01 | Volkswagen deliveries hit by chip shortages                                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173457892   | col_date=2020-10-01/col_int=11   | uuid:10            | 20220328173457892_0_22 | 6b9bf553-bd02-42f1-b5a4-014b207b587e-0_0-1008-1277_20220328173457892.parquet |
|    3 |           1 |       3 |        3 |     3.003 |     3.0003 |          30000000000.000000000300000000 | 2020-03-01 | Nation raises caution on overseas packages                                                                                                                                                                                                                                                                                | 1110001010101011001001 | 20220328173330230   | col_date=2020-03-01/col_int=3    | uuid:3             | 20220328173330230_0_11 | 99385896-04d6-4825-8fe8-6f11cc565118-0_0-568-512_20220328173330230.parquet   |
|    8 |           0 |       8 |        8 |     8.008 |     8.0008 |          80000000000.000000000800000000 | 2020-08-01 | Citi report finds MNC mood in China buoyant                                                                                                                                                                                                                                                                               | 1110001010101011001001 | 20220328173426990   | col_date=2020-08-01/col_int=8    | uuid:8             | 20220328173426990_0_11 | e999a63c-bf8e-420b-b91e-0c6f5bf5de8e-0_0-863-948_20220328173426990.parquet   |
|   13 |        NULL |    NULL |     NULL |      NULL |       NULL |                                    NULL | NULL       | NULL                                                                                                                                                                                                                                                                                                                      | NULL                   | 20220328173528828   | col_date=default/col_int=default | uuid:13            | 20220328173528828_0_17 | 978dd657-0c3d-47df-a512-136560c0c775-0_0-1158-1673_20220328173528828.parquet |
|   14 |        NULL |    NULL |     NULL |      NULL |       NULL | 12345678901234567890.123456789012345678 | NULL       |                                                                                                                                                                                                                                                                                                                           | NULL                   | 20220328173534438   | col_date=default/col_int=default | uuid:14            | 20220328173534438_0_23 | 978dd657-0c3d-47df-a512-136560c0c775-0_0-1187-1759_20220328173534438.parquet |
|    1 |           1 |       2 |        1 |     1.001 |     1.0001 |          10000000000.000000000100000000 | 2020-01-01 | Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?                                                                                                                                                                                                                                                     | 1110001010101011001001 | 20220328173315943   | col_date=2020-01-01/col_int=2    | uuid:1             | 20220328173315943_0_8  | 5736f8e9-c0d9-4d01-8c99-1f72e6dc31ba-0_0-483-428_20220328173315943.parquet   |
|    6 |           0 |       7 |        6 |     6.006 |     6.0006 |          60000000000.000000000600000000 | 2020-06-01 | The development of enterprises, as well as the push of local governments, has boosted the listings, the report said. For example, in early 2018, Central Chinas Hubei province launched a plan to double its listed companies by including the listing of enterprises into the governments annual performance assessment. | 1110001010101011001001 | 20220328173408179   | col_date=2020-06-01/col_int=7    | uuid:6             | 20220328173408179_0_16 | 74828055-72ee-4aac-a574-db9f7dd5bacc-0_0-773-778_20220328173408179.parquet   |
|   11 |           1 |      12 |       11 |    11.011 |   11.00011 |         110000000000.000000000110000000 | 2020-11-01 | Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.                                                                                                                                                                                         | 1110001010101011001001 | 20220328173510429   | col_date=2020-11-01/col_int=12   | uuid:11            | 20220328173510429_0_14 | db560743-ab82-466d-aa1d-4b393a3dc5f4-0_0-1069-1429_20220328173510429.parquet |
|    2 |           0 |       3 |        2 |     2.002 |     2.0002 |          20000000000.000000000200000000 | 2020-02-01 | Xi Jinping, general secretary of the Communist Party of China Central Committee, called on Tuesday for unrelenting efforts in exercising full and rigorous governance over the Party, saying that the CPC will continue to show zero tolerance for corruption.                                                            | 1110001010101011001001 | 20220328173325236   | col_date=2020-02-01/col_int=3    | uuid:2             | 20220328173325236_0_10 | 480dbff0-d609-44e9-b390-7f0f44c9169a-0_0-539-482_20220328173325236.parquet   |
|    7 |           1 |       8 |        7 |     7.007 |     7.0007 |          70000000000.000000000700000000 | 2020-07-01 | Most listed companies in the top 10 cities come from industries such as information technology, electronics, mechanical equipment, medical biology, and electric power equipment, with information technology bearing most listed companies.                                                                              | 1110001010101011001001 | 20220328173421883   | col_date=2020-07-01/col_int=8    | uuid:7             | 20220328173421883_0_18 | 028eb4bd-34ed-412b-9fd0-97fbe2337199-0_0-834-890_20220328173421883.parquet   |
|   12 |           0 |      13 |       12 |    12.012 |   12.00012 |         120000000000.000000000120000000 | 2020-12-01 | The annual lantern show at Yuyuan Garden, a historic tourist site in Shanghai, will kick off on Tuesday and run through Feb 28.                                                                                                                                                                                           | 1110001010101011001001 | 20220328173522636   | col_date=2020-12-01/col_int=13   | uuid:12            | 20220328173522636_0_16 | f46da3af-bc0e-47f1-9c0d-4eaa709f548c-0_0-1129-1590_20220328173522636.parquet |
|    1 |           1 |       1 |        1 |     1.001 |     1.0001 |          10000000000.000000000100000000 | 2020-01-01 | Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?                                                                                                                                                                                                                                                     | 1110001010101011001001 | 20220328173307979   | col_date=2020-01-01/col_int=1    | uuid:1             | 20220328173307979_0_8  | 41a60295-8f3d-4510-92a1-cbadf878530e-0_0-456-404_20220328173307979.parquet   |
|    6 |           0 |       6 |        6 |     6.006 |     6.0006 |          60000000000.000000000600000000 | 2020-06-01 | The development of enterprises, as well as the push of local governments, has boosted the listings, the report said. For example, in early 2018, Central Chinas Hubei province launched a plan to double its listed companies by including the listing of enterprises into the governments annual performance assessment. | 1110001010101011001001 | 20220328173402362   | col_date=2020-06-01/col_int=6    | uuid:6             | 20220328173402362_0_10 | 0d63a4b7-aad9-4454-b551-b2de7dc86e3b-0_0-745-728_20220328173402362.parquet   |
|   11 |           1 |      11 |       11 |    11.011 |   11.00011 |         110000000000.000000000110000000 | 2020-11-01 | Light installations featuring the tiger, the Chinese zodiac animal for the upcoming lunar year, will be set up in the main plaza.                                                                                                                                                                                         | 1110001010101011001001 | 20220328173504921   | col_date=2020-11-01/col_int=11   | uuid:11            | 20220328173504921_0_13 | f22d70ad-303d-44a1-8ce7-c669f2c3e97b-0_0-1041-1354_20220328173504921.parquet |
|    2 |           0 |       2 |        2 |     2.002 |     2.0002 |          20000000000.000000000200000000 | 2020-02-01 | Xi Jinping, general secretary of the Communist Party of China Central Committee, called on Tuesday for unrelenting efforts in exercising full and rigorous governance over the Party, saying that the CPC will continue to show zero tolerance for corruption.                                                            | 1110001010101011001001 | 20220328173320684   | col_date=2020-02-01/col_int=2    | uuid:2             | 20220328173320684_0_9  | d843c0bd-335a-434d-be8d-a15637ce52bf-0_0-510-453_20220328173320684.parquet   |
|    7 |           1 |       7 |        7 |     7.007 |     7.0007 |          70000000000.000000000700000000 | 2020-07-01 | Most listed companies in the top 10 cities come from industries such as information technology, electronics, mechanical equipment, medical biology, and electric power equipment, with information technology bearing most listed companies.                                                                              | 1110001010101011001001 | 20220328173416623   | col_date=2020-07-01/col_int=7    | uuid:7             | 20220328173416623_0_17 | 7b0632a8-bef3-4822-a23a-cd7d87115e48-0_0-805-834_20220328173416623.parquet   |
|   12 |           0 |      12 |       12 |    12.012 |   12.00012 |         120000000000.000000000120000000 | 2020-12-01 | The annual lantern show at Yuyuan Garden, a historic tourist site in Shanghai, will kick off on Tuesday and run through Feb 28.                                                                                                                                                                                           | 1110001010101011001001 | 20220328173517108   | col_date=2020-12-01/col_int=12   | uuid:12            | 20220328173517108_0_15 | a6b2b7ff-3a33-439b-b581-acc6aade96f9-0_0-1100-1509_20220328173517108.parquet |
+------+-------------+---------+----------+-----------+------------+-----------------------------------------+------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------+---------------------+----------------------------------+--------------------+------------------------+------------------------------------------------------------------------------+
26 rows in set (0.07 sec)
```

### Expected behavior (Required)
return the right result

### Real behavior (Required)
query error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 branch-master
```
mysql> select current_version();
+----------------------+
| current_version()    |
+----------------------+
| MAIN-RELEASE 96e57c1 |
+----------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by DorianZheng**
Signed-off-by: dorianzheng <xingzhengde72@gmail.com>

## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/6028  https://github.com/StarRocks/starrocks/issues/7177  https://github.com/StarRocks/starrocks/issues/7330 

## Problem Summary(Required) ：

Since `DataStreamRecvr::SenderQueue::_build_chunk_meta` is only called for the first chunk it receive, so the subsequent chunk should remain the same column order as the first chunk, otherwise, the deserialize might get wrong   type from chunk meta.

### CODE REVIEW

**Comment by DorianZheng** [comment]
@stephen-shelby @dirtysalt @miomiocat @caneGuy PTAL

**Comment by miomiocat** [comment]
LGTM

File: be/src/exec/vectorized/hdfs_scanner.cpp [hunk 1]
```diff
@@ -213,7 +213,20 @@ void HdfsScannerContext::set_columns_from_file(const std::unordered_set<std::str
...
+    ck->set_num_rows(row_count);
+    for (auto* slot_desc : not_existed_slots) {
+        auto col = ck->get_column_by_slot_id(slot_desc->id());
+        if (row_count > 0) {
+            col->append_default(row_count);
+        }
+    }
+}
+
+void HdfsScannerContext::append_not_existed_columns_to_chunk(vectorized::ChunkPtr* chunk, size_t row_count) {
```
Comment (dirtysalt) [inline be/src/exec/vectorized/hdfs_scanner.cpp:229]: `update` ?

File: be/src/exec/vectorized/hdfs_scanner.cpp [hunk 2]
```diff
@@ -232,7 +245,7 @@ StatusOr<bool> HdfsScannerContext::should_skip_by_evaluating_not_existed_slots()
 
     // build chunk for evaluation.
     ChunkPtr chunk = std::make_shared<Chunk>();
-    append_not_exised_columns_to_chunk(&chunk, 1);
+    append_not_existed_columns_to_chunk(&chunk, 1);
```
Comment (dirtysalt) [inline be/src/exec/vectorized/hdfs_scanner.cpp:248]: `update`?

File: be/src/exec/vectorized/hdfs_scanner.cpp (hunk 1 above)
Comment (DorianZheng) [reply be/src/exec/vectorized/hdfs_scanner.cpp:229]: no, this is another method, the update one is here: https://github.com/StarRocks/starrocks/pull/7413/files#diff-d87d2d93fc96f35903f3d611d309da8f4c2a675042ab62cfce0cba52e489e58fR216

Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scanner.cpp:229]: We still need `append_not_existed_columns_to_chunk` this method?

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

File: be/src/exec/vectorized/hdfs_scanner.cpp [hunk 3]
```diff
@@ -213,7 +213,21 @@ void HdfsScannerContext::set_columns_from_file(const std::unordered_set<std::str
     }
 }
 
-void HdfsScannerContext::append_not_exised_columns_to_chunk(vectorized::ChunkPtr* chunk, size_t row_count) {
+void HdfsScannerContext::update_not_existed_columns_to_chunk(vectorized::ChunkPtr* chunk, size_t row_count) {
```
Comment (caneGuy) [inline be/src/exec/vectorized/hdfs_scanner.cpp:216]: nit: for_chunk? and plz add some notes

**Review by caneGuy — COMMENTED** [review]
other LGTM

**Comment by DorianZheng** [comment]
@mergify backport branch-2.3

**Comment by DorianZheng** [comment]
@mergify backport branch-2.2

**Comment by DorianZheng** [comment]
@mergify backport branch-2.1

**Pull request merged**